src/Symfony/Component/Console/Helper/Tree.php

```
@@ -0,0 +1,118 @@
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @implements \RecursiveIterator<int, TreeNode>
+ */
+final class Tree implements \RecursiveIterator
+{
+    private readonly TreeStyle $style;
+
+    private \Iterator $childrenIterator;
+
+    public function __construct(
+        private readonly OutputInterface $output,
+        private readonly TreeNode $node,
+        ?TreeStyle $style = null,
+    ) {
+        $this->style = $style ?? TreeStyle::default();
+        $this->childrenIterator = new \IteratorIterator($node->getChildren());
+        $this->childrenIterator->rewind();
+    }
+
+    public function current(): TreeNode
+    {
+        return $this->childrenIterator->current();
+    }
+
+    public function key(): int
+    {
+        return $this->childrenIterator->key();
+    }
+
+    public function next(): void
+    {
+        $this->childrenIterator->next();
+    }
+
+    public function rewind(): void
+    {
+        $this->childrenIterator->rewind();
+    }
+
+    public function valid(): bool
+    {
+        return $this->childrenIterator->valid();
+    }
+
+    public function hasChildren(): bool
+    {
+        if (null === $current = $this->current()) {
+            return false;
+        }
+
+        foreach ($current->getChildren() as $child) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getChildren(): \RecursiveIterator
+    {
+        return new self($this->output, $this->current(), $this->style);
+    }
+
+    /**
+     * Recursively renders the tree to the output, applying the tree style.
+     */
+    public function render(): void
+    {
+        $treeIterator = new \RecursiveTreeIterator(
+            $this,
+            \RecursiveIteratorIterator::CHILD_FIRST,
+            \RecursiveIteratorIterator::CATCH_GET_CHILD,
+        );
```

```suggestion
        $treeIterator = new \RecursiveTreeIterator(
            $this
        );
```

I think only the first parameter is useful, "childfirst "mode see "child" before other tree node acting as "parent", I think it is not wanted : we want to keep the logical order "parent ->children" 

 [RecursiveIteratorIterator::CATCH_GET_CHILD](https://www.php.net/manual/en/class.recursiveiteratoriterator.php#recursiveiteratoriterator.constants.catch-get-child) will ignore exceptions thrown, same, I don't think it is needed